### PR TITLE
Support for val final modifier

### LINKED ISSUE #158: val causes intellij to show an invalid error when used alongside an anonymous class
val is supposed to infer the type of a value and make it final, so you should be able to use them in anonymous classes. In fact you can do this! Intellij doesn't believe you though:

    public class ValTest {
        public void doSomething() {
            val foo = new Foo();
            val bar = new Bar();
            foo.setBazListener(new BazListener () {
                @Override
                public Bar onBaz() {
                    return bar;
                }
            });
        }
    }

When using this code there is an error on ```bar``` in ```return bar``` that says "Variable bar is accessed from within inner class, needs to be declared final". I agree, however val indeed does make it final. This code actually compiles and runs fine. I'm currently using android studio 1.2.1.1, and I think it's based off of idea 14.

### PR DESCRIPTION
**Pull request by AlexejK**
Fixes #158 

Based on latest IDEA Trunk snapshot, the desired functionality is in place and allows `val` to be seen as final.

This PR does not cover `@Value` and similar issues, but is purely focused on `val`. For other issues I believe a small structural change might be needed.

PS: I've also updated the build files a bit to make it more flexible and also test against bleeding edge versions to spot possible failures.

### CODE REVIEW

File: src/main/java/de/plushnikov/intellij/plugin/processor/ValProcessor.java [hunk 1]
```diff
@@ -7,34 +7,20 @@
...
-import com.intellij.psi.PsiForeachStatement;
-import com.intellij.psi.PsiJavaCodeReferenceElement;
-import com.intellij.psi.PsiLambdaExpression;
-import com.intellij.psi.PsiLocalVariable;
-import com.intellij.psi.PsiNewExpression;
-import com.intellij.psi.PsiParameter;
-import com.intellij.psi.PsiReferenceParameterList;
-import com.intellij.psi.PsiType;
-import com.intellij.psi.PsiTypeElement;
+import com.intellij.psi.*;
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/processor/ValProcessor.java]: Please change your CodeStyleSettings, I will not use any (*) -imports in the project.

File: src/main/java/de/plushnikov/intellij/plugin/processor/ValProcessor.java [hunk 2]
```diff
@@ -104,6 +90,17 @@ private static boolean isSameName(String className) {
     return LOMBOK_VAL_SHORT_NAME.equals(className) || LOMBOK_VAL_FQN.equals(className);
   }
 
+  public Boolean hasModifierProperty(@NotNull PsiModifierList modifierList, @NotNull String name) {
+    final PsiElement parent = modifierList.getParent();
+
+    //Only apply final modifier to val local variables
+    if (parent instanceof PsiLocalVariable && isVal((PsiLocalVariable) parent)) {
+      return PsiModifier.FINAL.equals(name) ? Boolean.TRUE : null;
```
Comment (mplushnikov) [inline src/main/java/de/plushnikov/intellij/plugin/processor/ValProcessor.java]: It could be better to check for PsiModifier.FINAL.equals(name) first, before checking for instanceof and isVal

**Comment by mplushnikov** [comment]
Thank you for PR! I just have commented some points...

**Comment by AlexejK** [comment]
Thanks, Updated!

**Comment by AlexejK** [comment]
Damn it.. It seems to have reformatted the whole file.. Sorry! Will fix it
